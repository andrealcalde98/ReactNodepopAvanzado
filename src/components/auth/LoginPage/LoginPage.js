import { connect } from 'react-redux';
import T from 'prop-types';
import { useState, useMemo } from "react";
import FormField from "../../common/FormField.js";
import Button from "../../common/Button";

import { authLogin, uiResetError } from '../../../store/action'
import { getUi } from '../../../store/selectors';

import "./LoginPage.css";

function LoginPage({ onLogin, resetError, isLoading, error }) {
    const [value, setValue] = useState({ email: "", password: "" });
    const [checked, setIsChecked] = useState(null);

    const handleChange = (event) => {
        setValue((prevState) => ({
            ...prevState,
            [event.target.name]: event.target.value,
        }));
    };

    const handleCheck = (event) => {
        console.log(event.target.checked)
        setIsChecked(event.target.checked)
    }

    const handleSubmit = async event => {
        event.preventDefault();
        await onLogin(checked, value).then(() => {
        });
    };

    const buttonDisabled = useMemo(
        () => isLoading || !value.email || !value.password,
        [isLoading, value.email, value.password]
    );

    return (
        <div className="loginPage">
            <h1 className="loginPage-title">Log in to NodePop</h1>
            <form className="loginForm" onSubmit={handleSubmit}>
                <FormField
                    type="text"
                    name="email"
                    label="Email"
                    className="loginForm-field"
                    value={value.email}
                    onChange={handleChange}
                    autofocus
                />
                <FormField
                    type="password"
                    name="password"
                    label="password"
                    className="loginForm-field"
                    value={value.password}
                    onChange={handleChange}
                />
                <Button
                    className="loginForm-submit"
                    type="submit"
                    variant="primary"
                    disabled={buttonDisabled}
                >
                    Log in
                </Button>
                <br></br>
                <label>
                    <input type="checkbox" id="cbox1" onChange={handleCheck} />
                    Quieres mantener la sesi√≥n iniciada?
                </label>
            </form>
            {error && (
                <div onClick={resetError} className="loginPage-error">
                    {error.message}
                </div>
            )}
        </div>
    );
}

LoginPage.propTypes = {
    onLogin: T.func.isRequired,
};

// const ConnectedLoginPage = (props) => (
//     <AuthContextConsumer>
//         {(auth) => <LoginPage onLogin={auth.handleLogin} {...props} />}
//     </AuthContextConsumer>
// );

const mapStateToProps = state => {
    return getUi(state);
};

const mapDispatchToProps = {
    onLogin: authLogin,
    resetError: uiResetError,
};

const ConnectedLoginPage = connect(
    mapStateToProps,
    mapDispatchToProps,
)(LoginPage);

export default ConnectedLoginPage;