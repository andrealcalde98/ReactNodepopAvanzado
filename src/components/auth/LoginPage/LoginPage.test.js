import { fireEvent, render } from '@testing-library/react';
import { LoginPage } from './LoginPage';

describe('LoginPage', () => {
  test('snapshot', () => {
    const { container } = render(<LoginPage onLogin={() => { }} />);
    expect(container).toMatchSnapshot();
  });

  test('should reset error', () => {
    const error = { message: 'Netwrok error' };
    const resetError = jest.fn();
    const onLogin = () => { };

    const { getByText } = render(
      <LoginPage onLogin={onLogin} error={error} resetError={resetError} />,
    );

    const errorField = getByText(error.message);
    fireEvent.click(errorField);
    expect(resetError).toHaveBeenCalled();
  });
});
