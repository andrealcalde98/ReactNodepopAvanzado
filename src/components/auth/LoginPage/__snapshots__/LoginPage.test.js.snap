// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage snapshot 1`] = `
<div>
  <div
    class="loginPage"
  >
    <h1
      class="loginPage-title"
    >
      Log in to NodePop
    </h1>
    <form
      class="loginForm"
    >
      <div
        class="formField formField--focused loginForm-field"
      >
        <label
          class="formField-label"
        >
          <span>
            Email
          </span>
          <input
            autocomplete="off"
            class="formField-input"
            name="email"
            type="text"
            value=""
          />
        </label>
      </div>
      <div
        class="formField loginForm-field"
      >
        <label
          class="formField-label"
        >
          <span>
            password
          </span>
          <input
            autocomplete="off"
            class="formField-input"
            name="password"
            type="password"
            value=""
          />
        </label>
      </div>
      <button
        class="sc-bdvvtL bzLRQE loginForm-submit"
        disabled=""
        type="submit"
      >
        Log in
      </button>
      <br />
      <label>
        <input
          id="cbox1"
          type="checkbox"
        />
        Quieres mantener la sesi√≥n iniciada?
      </label>
    </form>
  </div>
</div>
`;
